src/main/java/com/cricket/project/service/impl/TeamServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package com.cricket.project.service.impl;
+
+import com.cricket.project.enums.PlayerType;
+import com.cricket.project.exception.TeamException;
+import com.cricket.project.model.Player;
+import com.cricket.project.model.Team;
+import com.cricket.project.repository.PlayerRepository;
+import com.cricket.project.repository.TeamRepository;
+import com.cricket.project.service.TeamService;
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Data
+public class TeamServiceImpl implements TeamService {
+
+    @Autowired
+    private PlayerRepository playerRepository;
+    @Autowired
+    private TeamRepository teamRepository;
+
+    private Map<Integer, Integer> exist = new HashMap<>();
+    int count=0;
+    @Override
+    public Team createTeam(Map<String,String> teamName) throws TeamException {
+        count++;
+        if (count >= 3) {
+            throw new TeamException("Max 2 teams allowed!");
+        }
+        List<Integer> playersInTeam = setPlayersInTeam();
+
+        Team team = Team.builder()
+                .id(count)
+                .teamName(teamName.get("teamName"))
+                .teamCaptain(playerRepository.findPlayerById(playersInTeam.get(0)).getPlayerName())
+                .teamPlayersId(playersInTeam)
+                .totalWins(0)
+                .totalMatchesPlayed(0)
+                .build();
+
+        teamRepository.saveTeam(team);
+        return team;
+
+    }
+
+    public List<Integer> setPlayersInTeam()
+    {
+        List<Player> bowlers = playerRepository.findPlayersByPlayerType(PlayerType.valueOf("BOWLER"));
+        List<Player> batsmen = playerRepository.findPlayersByPlayerType(PlayerType.valueOf("BATSMAN"));
+        List<Player> allRounders = playerRepository.findPlayersByPlayerType(PlayerType.valueOf("ALLROUNDER"));
+        List<Integer> playersInTeam = new ArrayList<>();
+
+        int numberOfBowlers = 4;
+        int numberOfBatsman = 4;
+        int numberOfAllRounder = 3;
+
+        setPlayersInTeamByPlayerType(allRounders, playersInTeam, numberOfAllRounder);
+        setPlayersInTeamByPlayerType(batsmen, playersInTeam, numberOfBatsman);
+        setPlayersInTeamByPlayerType(bowlers, playersInTeam, numberOfBowlers);
+        return playersInTeam;
+    }
+
+    private void setPlayersInTeamByPlayerType(List<Player> players, List<Integer> playersInTeam, int numberOfPlayerType) {
+        for(Player player:players)
+        {
```

Comment:
we prefer keeping the starting curly brace to be in line with the for loop/if statements